nifi-nar-bundles/nifi-cdc/nifi-cdc-mssql-bundle/nifi-cdc-mssql-processors/src/main/java/org/apache/nifi/cdc/mssql/processors/CaptureChangeMSSQL.java

```diff
@@ -0,0 +1,387 @@
+    public static final String INITIAL_TIMESTAMP_PROP_START = "initial.timestamp.";
+
+    public static final PropertyDescriptor RECORD_WRITER = new PropertyDescriptor.Builder()
+            .name("record-writer")
+            .displayName("Record Writer")
+            .description("Specifies the Controller Service to use for writing out the records")
+            .identifiesControllerService(RecordSetWriterFactory.class)
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor DBCP_SERVICE = new PropertyDescriptor.Builder()
+            .name("cdcmssql-dbcp-service")
+            .displayName("Database Connection Pooling Service")
+            .description("The Controller Service that is used to obtain connection to database")
+            .required(true)
+            .identifiesControllerService(DBCPService.class)
+            .build();
+
+    public static final PropertyDescriptor CDC_TABLES = new PropertyDescriptor.Builder()
+            .name("cdcmssql-cdc-table-list")
+            .displayName("CDC Table List")
+            .description("The comma delimited list of tables in the source database to monitor for changes. If no tables "
+                    + "are specified the [cdc].[change_tables] table is queried for all of the available tables with change tracking enabled in the database.")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TAKE_INITIAL_SNAPSHOT = new PropertyDescriptor.Builder()
+            .name("cdcmssql-initial-snapshot")
+            .displayName("Generate an Initial Source Table Snapshot")
+            .description("Usually CDC only includes recent historic changes. Setting this property to true will cause a snapshot of the "
+                + "source table to be taken using the same schema as the CDC extracts. The snapshot time will be used as the starting point "
+                + "for extracting CDC changes.")
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor FULL_SNAPSHOT_ROW_LIMIT = new PropertyDescriptor
+            .Builder().name("cdcmssql-full-snapshot-row-limit")
+            .displayName("Change Set Row Limit")
+            .description("If a very large change occurs on the source table, "
+                    + "the generated change set may be too large too quickly merge into a destination system. "
+                    + "Use this property to set a cut-off point where instead of returning a changeset a full snapshot will be generated instead. "
+                    + "The fullsnapshot attribute will be set to true when this happens.")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
```

Comment:
Consider adding Expression Language support here, the limit might be different between dev and production systems for example. Also there's a typo in the description, should read "too large to quickly merge", and the term "change set" or "changeset" should be used consistently (I'm fine with either) throughout the doc.

Reply:
Settled on "changeset". Will update.